### PR TITLE
Add tests...

### PR DESCRIPTION
**Pull request by rkurchin**
Hoping this will create a good baseline for @DhairyaLGandhi to troubleshoot the AD. A work in progress atm, hope to finish this week...

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
Really Excited to see this come through! Could we document where the ground truth numbers for the BV tests came from? Maybe we need to have reproducer scripts for them (at a later point) but it would be good to know how those numbers were procured just so someone coming across the repo can backsolve to the expected/ reference behaviour as they develop on this repo further.

**Comment by rkurchin** [comment]
Do you just mean to add comments doing out the math that got me to those numbers? It follows pretty straightforwardly from the BV equation...

File: test/fitting_tests.jl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    # test a number less than the initial guess
+    V1 = 0.05
+    k1 = bv(V1)
+    @test isapprox(fit_overpotential(bv, k1)[1], V1, atol=1e-5)
+    # and one greater
+    V2 = 0.15
+    k2 = bv(V2)
+    @test isapprox(fit_overpotential(bv, k2)[1], V2, atol=1e-5)
+    # test that vectorized fitting works...this test currently fails
+    @test isapprox.(fit_overpotential(bv, [k1, k2]), [V1, V2], atol=1e-5)
```
Comment (DhairyaLGandhi) [inline test/fitting_tests.jl:12]:
```suggestion
    @test isapprox(fit_overpotential(bv, [k1, k2]), [V1, V2], atol=1e-5)
```

**Comment by DhairyaLGandhi** [comment]
My comment was to make sure we are able to track what we are testing. Whether by pointing users to the origin of reference values or a reference implementation.

File: test/fitting_tests.jl [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+end
+
+@testset begin "Model Parameter Fitting" begin
```
Comment (DhairyaLGandhi) [inline test/fitting_tests.jl:15]:
```suggestion
@testset "Model Parameter Fitting" begin
```

**Comment by rkurchin** [comment]
> My comment was to make sure we are able to track what we are testing. Whether by pointing users to the origin of reference values or a reference implementation.

Sorry, I'm still not completely sure what you mean by this. All I did was analytically invert the equations that specify the model. Should I just say that in a comment? I suspect that's not really sufficient for what you're envisioning...

**Comment by DhairyaLGandhi** [comment]
Yeah a simple comment should be sufficient for now. We sometimes add simple breakdowns for future reference https://github.com/FluxML/Flux.jl/blob/master/test/layers/normalisation.jl, but it isn't a strict necessity.

**Comment by rkurchin** [comment]
Note mostly for me: still to do here:

- [ ] add comments per Dhairya's suggestion above clarifying where "ground truth" numbers come from in all relevant cases
- [ ] add kinetic g and µ tests for models other than `ButlerVolmer`
- [ ] add tests for `common_tangent` and `find_phase_boundaries` functions

Once I finish all these, I'll be happy to merge this, even though there will be a few lingering todos in terms of more full coverage.

**Pull request merged**